### DISCUSSION TITLE
Mark unwatched behavior on play

### DISCUSSION THREAD

**Discussion by SenorSmartyPants** [jellyfin/jellyfin-meta#56] [Ideas]
I think the current behavior of marking resumable content unwatched on play is wrong. I have created an issue  jellyfin/jellyfin#7535 and PRs to give options to alter this behavior  jellyfin/jellyfin#8275 or jellyfin/jellyfin-web#3805 . 

Now it seems popular opinion might be switching to not mark items as unwatched when play has started. 

Please check out my issue and discuss here.

**Comment by nielsvanvelzen** [comment]
Please make sure to specify all information in the meta discussion instead of linking to other places. 

Without reading the linked threads my initial reaction is that I agree. An item should be marked watched when completed only and marked unwatched only when playback is stopped and the resume functionality enabled (I believe this is an absolute value depending on the item type).

**Reply by SenorSmartyPants** [reply]
IMO nothing should be marked as unwatched based on playing an item. Item is unwatched when imported into JF. Could be marked as watched by a history import plugin like Trakt. 

But if you have watched an item, JF should never automatically mark it unwatched. Manually by a user, absolutely.

**Reply by klueman** [reply]
You said you agree but then the rest of your comment sounds like maybe you misunderstand what the complaint is so to clarify:

>An item should be marked watched when completed only

Agreed, if it's new content that has not been completed yet.

>and marked unwatched only when playback is stopped and the resume functionality enabled

Agreed in cases where the content has not been marked as watched previously.

If it has been watched once already, the unwatched status _overwrites_ the watched status until it's completed again or manually marked completed.

This is not desirable functionality in certain cases, because the watched status affects other things in Jellyfin beyond triggering the resume functionality. The unwatched tag is serving a duel role here as both an indicator that playback was interrupted and may need to be resumed, but it's *also* serving as the equivalent of a "NEW" tag for certain functions of Jellyfin, and the UI reflects that. But content being re-watched is not "new" and people may not want Jellyfin to display it as if it is.

There should be a toggle or lock or something that allows certain content to remain permanently "watched".

**Comment by FOSSilizedDaemon** [comment]
I absolutely think this is a needed change. In fact, I found this trying to figure out how to fix what I thought was a bug. This issue makes it hard to take advantage of other features such as playlists or shuffle play because re-watching a series causes all of my watched progress to be removed. I think this is heavily needed.

Edit:
I do not think we are kn the same page. Jellyfin should just handle watched tracking how all major streaming platforms do: when you watch something it tracks to what point you stopped, if you are in the later end of the episode it is marked as watched. Currently the issue with Jellyfin is that rewatching content causes entire shows to be marked unwatched due to some bug. Having to manually manage watched/unwatched would be tiring and having to use a whole new platform like trakt would be irritating.

**Comment by davemsh** [comment]
JF instantly marking watched media as unwatched the moment it starts playing is extremely irritating, since it results in my library getting out of sync with what's in Trakt.tv.

Why not have a toggle setting to override this behavior? Imo the only time any watched items should be switched back to the unwatched state is if the user explicitly unchecked the watched status. No other reasons.

**Comment by dan-riley** [comment]
Please make this change.  I'm testing out a few things (which I tend to do a lot), playing something briefly then stopping, and it's totally nuking my watches.  Now I have to make sure I use files that have never been played before so I can make sure I can clear their history.  It also caused an infinite loop with Trakt where it and Jellyfin kept resetting each other each time a sync was performed.  It only stopped when I manually changed Jellyfin to match Trakt (ie, Unwatched, since there's no way I can find to manually set a Watched date).

**Comment by klueman** [comment]
Would greatly love to see this implemented. I see there might be some confusion as to why so to explain:

As an example, the shuffle function. Say you like putting a certain show you've already watched once on shuffle, but occasionally skip certain episodes when they come up. Those episodes might lose their watched status in the library and need to be manually set back to watched. Or maybe you felt like re-watching a certain episode of something one night but didn't finish it, and have no intention to go back and finish it. You've already seen it once, it was just something to put on, you don't need to finish that episode again.

In those cases, with Jellyfin, those episodes you were rewatching now flood the "Continue Watching" section, and have counts for unwatched episodes on their show's images. To clean that up you have to go back in and manually mark each episode/show watched, which is tedious. It also affects interactions with Trakt.

This is not an unusual manner for people to use their media server, and in this way, the video content is being treated much the same way you would expect audio content to work in a music app. Songs don't get marked unplayed or "new" because you hit play, the played count just increases by 1.

Basically what I'd like to see is a lock on the watched status that can be toggled for certain content. "This has been watched to completion and should never be marked unwatched again regardless of activity." Or simply treat the Watched status like the Read status on email. It gets marked Read when opened, but it only ever gets marked Unread when you do it manually.

I'm aware the watched status is tied to the resume function but that feels like part of the issue. The "watched" tag shouldn't be tied to the playback progression in such a way that it must be overwritten each time a new playback session is started. There's a difference between "I have never watched this" and "I'm in process of watching this a second time" but the current usage of the unwatched tag treats them as the same thing.

Podcast apps are another good example. Even if playback position is retained on a re-listen of an old episode, that episode is not flagged as "new" or "unwatched". It just saved the position, didn't change the status.

**Comment by journeym** [comment]
Omg ppl you shoud have fixed it already, any sane person understands that this is how that should be. If you unsure copy form plex, ppl do everything right.

**Comment by cewert** [comment]
I despise this behavior and would welcome any solution whether that be a user setting or changing default behavior.

What are we waiting on as far as making a decision? Is there anything we can do to help move this forward?

**Comment by davemsh** [comment]
This problem is still as irritating as ever, and it doesn't sound like anyone is actually opposed to this. Why was the original PR to change this behavior from two years ago closed and directed to here for a discussion in the first place?

**Comment by gbr412** [comment]
This is still an issue three years after it was initially brought to attention. Please change this behavior or provide a better solution than having to manually mark everything I rewatch for a couple of minutes as watched again.

**Comment by xelu** [comment]
[Reddit thread](https://www.reddit.com/r/jellyfin/comments/1piyfja/does_anyone_really_think_this_is_how_unwatching/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)

**Comment by lkeels** [comment]
After three years with no attention, it's time to address this.

**Reply by nielsvanvelzen** [reply]
There was in fact a new pull request opened earlier today so it is likely something will happen. Still unfortunate the original author refused to create a proper proposal though.

**Reply by ranaldsgift** [reply]
I guess it's more unfortunate that this has been left to sit for years rather than the original author not creating a "proper" proposal.

**Reply by nielsvanvelzen** [reply]
Feel free to pick it up then!

**Reply by srcrist** [reply]
> I guess it's more unfortunate that this has been left to sit for years rather than the original author not creating a "proper" proposal.

As always, Jellyfin is developed on an entirely volunteer basis. People work on the things that interest them in their spare time. There is no team of people to magically assemble to work on other people's preferences. If you want a change, you either submit a PR or find someone willing to write it for you. We are not a company. We do not pay people to spend time working on other people's desired changes. There is no alternative option here.

**Reply by journeym** [reply]
what's your definition of "proper proposal"? OP literally created 2 PRs with a working solution

**Reply by nielsvanvelzen** [reply]
In some cases we (the Jellyfin team) think a change will need some discussion in how to solve a problem or work on a big new thing. In those cases we use "proposals" in the form of GitHub discussions in this exact repository. Sometimes we'll ask someone who created a pull request (or plans to create one) to first create a proposal so we can have a discussion without cluttering the pull request itself.

In this specific case, a pull request was made that was a quite complicated solution. Therefor the team asked for a proposal. The content of a proposal will always differ depending on what is desired. In this case I'd expect answers to 2 questions: what is the problem? What is the **proposed** solution for the problem?

Then we could've had the discussion about the proposed solution so we (the Jellyfin team) are all on board with it.

What happened this time though, is that a lazy discussion was opened which just links to the already made pull request with a "figure it out" attitude. When asked to create a more elaborate proposal it was ignored. Until today when apparently someone brought this issue up on Reddit, and someone else already made a pull request with a much more simple solution. Although, a good proposal would still be nice as there might be a need for a more elaborate behavior change instead of the current "never mark as unwatched" implementation.

**Reply by SenorSmartyPants** [reply]
Here's my [original PR](https://github.com/jellyfin/jellyfin/pull/8247#issuecomment-1212584563) (which is exactly the same as the new one submitted recently, and approved). 

I was told by JF team member to make it configurable, and so new PR. But that wasn't clear enough so, start a discussion...

I've made several PRs to JF over the years, some merged some not. Anything more than a couple lines of code can become quite the process to get review and approved. Maybe I am a terrible developer and write terrible code. But telling non team contributors they are lazy and don't do things properly is not a way to encourage continued contribution.

**Reply by srcrist** [reply]
To be clear: the issue was not a lazy PR. The issue was that this very thread was never expanded to provide the platform for discussion that was requested. Nobody wants to have to click links around the repo to understand the problem for discussion. You were simply asked to consolidate the proposal here, and declined to do so.

**Reply by klueman** [reply]
The problem is this is all coming off very disingenuous. Expecting people to click on links that don't even take you off Git is not that big of an ask.

If the issue was the original proposal wasn't posted here in its entirety, and that **alone** is the reason why no one responded to any of the other comments, why is the team responding today, clarifying why the discussion never happened after it gets posted to reddit?

I made two long, detailed, good faith comments laying out the concerns with the function, citing examples of where it causes issues, and examples of apps with similar functions to what was being requested. I made them *explicitly* because I wanted to be absolutely sure there was substance here to start a discussion, and that the team was understanding where people were coming from. Both of them were ignored.

And yet, someone makes a condescending, snarky comment, the link is shared with reddit, and you'll respond to that?

The point is, yeah, SmartyPants could have followed instructions better, but it also just doesn't feel anyone wanted to help get the PR merged either, or give anyone any guidance on what they could change about it.

**Reply by srcrist** [reply]
I mean, it does not seem unreasonable that people are responding when the topic is raised again? Confusion on that point seems, to me, to be based on an entirely mistaken opinion that the issue was being ignored in the first place.

The reason that further discussion was requested is because there is not, in fact, consensus about the issue among maintainers. This was intended functionality implemented by Emby. It is not, in fact, technically a bug--at least in some stakeholders eyes. That's not to say that people are opposed to changing it. There seems to be a pretty broad consensus that, sure, this is probably not an ideal behavior. People just wanted to have a space to discuss the issue further so the specific approach to solving the problem could be clarified in public. The project might prefer a submission that makes this a user configurable option, for example, or there may be other components that rely on the behavior as originally developed, and we might need to require a broader PR. Establishing these things is the purpose of the discussion. 

A reasonable request was made that the author expand and consolidate the discussion in this space. The project has a right to make such requests. Different projects and their maintainers may have different approaches, but that was the request made here to facilitate discussion. The presumption in this community is that the author is the primary stakeholder who bears the responsibility of facilitating the communication and development around their proposed changes. Seems reasonable enough to me, personally. When the author did not do that, the discussion sort of petered out. I get that this is frustrating, but nobody is being disingenuous about it. There's a reason that both the discussion and the consolidation were requested: to make this discussion easier and more comprehensive in order to arrive  at a solution.

Your personal contribution here seems fine. You added good detail. I would imagine, unfortunately, that it probably wasn't seen by a lot of people, many of whom were probably waiting for the requested additional details to be consolidated in this thread. An admittedly subjective request, but that's the request that was made.

We're now having this conversation several years later. Some perspectives may now have changed. Some stakeholders may have changed, even. Hell, maybe people no longer feel that a conversation is necessary. I don't know. I can't speak for everyone. But I don't think it's anyone's job on "the team" to help someone get a PR merged. They have a ton of other things to do--both for this project and elsewhere. We generally rely on the contributor to facilitate making it easier to discuss and review their changes alongside the many others. That's the thing that did not happen here.

**Comment by lkeels** [comment]
The excuses for this not being fixed are wild. And it's not a feature
request. It's obviously something that should never have been the way it
is. In my view, it's a bug that needs to be fixed.

**Reply by srcrist** [reply]
There is no excuse. There is simply nobody obligated to fix it. That's just not how the development process in this repository works. There's nobody for you to blame. There was code that some people wanted changed, a proposal was requested for discussion of that change, and that proposal was never drafted.

To be clear: Even if it *is* a bug to be fixed, there is no specific person or group of people whose responsibility it is to fix every bug. It's that simple. The people who put this code in there work for Emby. You could yell at them, I suppose, if you pay them for it. But it has nothing to do with the code base for Jellyfin, which is maintained and fixed on an entirely volunteer basis. Some of those volunteers requested a discussion of the proposed change, and that discussion never happened, and nobody made any progress. Such is life in FOSS.

**Reply by maru801** [reply]
> The excuses for this not being fixed are wild. And it's not a feature request. It's obviously something that should never have been the way it is. In my view, it's a bug that needs to be fixed.

I disagree with it being a bug. It was a deliberate design choice Emby used and that Jellyfin forked.
The current next-up behavior works because of how the watch indicator currently works.

From my view, the current watched indicator is doing "double-duty" in that it's doing two functions at once.
There should probably be a separate watched indicator that is permanent and stays on unless the user removes it.
The current behavior is also incorporating the ability to have it show rewatch progress which conflicts with the indicator showing if something was watched in the past.

It's going to take some more discussion about how to rework the system to make both still work in Jellyfin. That's not a fast process, especially if you see what got Jellyfin stuck on release candidate testing for close to 8 months this year for v10.11.

There are other things too that people view as being neglected due to how much time has gone by without nothing being done.
Like one of them is the redesign of the webUI (which I believe has been discussed for a few years now without any major changes).
There isn't really any indicator to which features are more important to fix, so what usually happens is more work is done on the fronts volunteers feel like working on and that impact them the most.

**Comment by lkeels** [comment]
So you want donations, but no obligations. Got it.

**Reply by srcrist** [reply]
Do you understand what the word "donation" means?

**Reply by srcrist** [reply]
Also, please stop replying here with the complaints about project administration. That's not the purpose of this discussion and we have sufficiently and clearly addressed our position on both the actual topic and this discussion as a whole. You are free not to donate or use the software if you dislike the development or financial practices. Please keep further responses to the topic of this proposed change and any code or UX related concerns or feedback you may have.

**Comment by lkeels** [comment]
It means paying money. It doesn't matter where it goes. It still means I
paid money.
